build_tools/hack/env_check/check_tools.py

```diff
@@ -0,0 +1,875 @@
+from __future__ import annotations
+from typing import Literal, Optional, Union, Tuple
+from types import EllipsisType, NoneType
+import os, re, sys, shutil
+from pathlib import Path
+from abc import ABC
+from .utils import cstring, Emoji
+from .find_tools import *
+from .device import Device
+
+
+def msg_stat(status: Literal["pass", "warn", "err"], program: str, message: str):
+    if isinstance(program, find_program):
+        match status:
+            case "pass":
+                _emoji = Emoji.Pass
+            case "warn":
+                _emoji = Emoji.Warn
+            case "err":
+                _emoji = Emoji.Err
+
+        return f"[{_emoji}][{program}] {message}"
+
+    elif isinstance(program, str):
+        match status:
+            case "pass":
+                _emoji = Emoji.Pass
+            case "warn":
+                _emoji = Emoji.Warn
+            case "err":
+                _emoji = Emoji.Err
+
+        return f"[{_emoji}][{program}] {message}"
+
+
+class Check_List:
+    def __init__(self, checks: list):
+        self.todo = checks
+        self.passed, self.warned, self.errs = 0, 0, 0
+        self.check_record = []
+
+    def run(self):
+        for each_test in self.todo:
+            _stat, _except, _result = each_test.__call__()
+            print(f"{_stat} {_except}")
+            self.check_record.append(_result)
+
+    @property
+    def summary(self):
+        self.run()
+        pass_num = cstring(self.check_record.count(True), color=(55, 255, 125))
+        warn_num = cstring(self.check_record.count(False), color="warn")
+        err_num = cstring(self.check_record.count(None), color="err")
+
+        print(
+            f"""
+                                Compoments check {pass_num} Passed, {warn_num} Warning, {err_num} Fatal Error"""
+        )
+
+
+# Program Checkers.
+
+
+class check_program(ABC):
```

Comment:
class naming for a class
```suggestion
class CheckProgram(ABC):
```